### PR TITLE
Remove old deprecated extensions.

### PR DESCRIPTION
**Pull request by Carreau**
All theses extensions are already in their own repository.

Local Rmagic could in any cases not be imported as %load_ext would raise
if rpy2 not available.

Cython and sympy where still trying to do the right things, which I
removed, and converted to warnings. (cannot use DeprecationWarning as
they would be hidden) and no way to get the right stacklevel as this is
magic that load the ext.

### CODE REVIEW

File: IPython/extensions/cythonmagic.py [hunk 1]
```diff
@@ -8,36 +10,17 @@
...
 
 ## still load the magic in IPython 3.x, remove completely in future versions.
 def load_ipython_extension(ip):
     """Load the extension in IPython."""
     
-    print("""The Cython magic has been moved to the Cython package, hence """)
-    print("""`%load_ext cythonmagic` is deprecated; please use `%load_ext Cython` instead.""")
+    warnings.warn("""
+The Cython magic has been moved to the Cython package, hence %load_ext cythonmagic` is deprecated; 
+Please use `%load_ext Cython` instead.""")
```
Comment (minrk) [inline IPython/extensions/cythonmagic.py]: It's not deprecated if it doesn't work anymore. I think this is fine if you just remove the "hence..." clause.

File: IPython/extensions/rmagic.py [hunk 2]
```diff
@@ -1,702 +1,13 @@
...
-                R_DOC = dedent(RMagics.R.__doc__),
-                RPUSH_DOC = dedent(RMagics.Rpush.__doc__),
-                RPULL_DOC = dedent(RMagics.Rpull.__doc__),
-                RGET_DOC = dedent(RMagics.Rget.__doc__)
-)
-
-
 def load_ipython_extension(ip):
     """Load the extension in IPython."""
     warnings.warn("The rmagic extension in IPython is deprecated in favour of "
```
Comment (minrk) [inline IPython/extensions/rmagic.py]: s/is deprecated in favour of/has moved to/

**Comment by Carreau** [comment]
Fixed comments

File: IPython/extensions/sympyprinting.py [hunk 3]
```diff
@@ -17,154 +19,14 @@
...
-        latex_formatter.for_type_by_name(
-            'sympy.matrices.matrices', 'Matrix', print_latex
-        )
-
-        for cls in printable_containers:
-            # Use LaTeX only if every element is printable by latex
-            latex_formatter.for_type(cls, print_latex)
-
-        _loaded = True
+    warnings.warn("The sympyprinting extension in IPython is deprecated, "
```
Comment (minrk) [inline IPython/extensions/sympyprinting.py]: missed one more no-longer deprecated.

Comment (Carreau) [inline IPython/extensions/sympyprinting.py]: Done.

**Pull request merged**